### ISSUE TITLE
HK has not yet implemented Sat shift

### ISSUE THREAD

**Issue by greatfung** [peopledoc/workalendar#477]
# Boxing day & XMas shift
        shifts = self.shift_christmas_boxing_days(year=year)
        days.extend(shifts)

so for 2020, Dec 28 (Mon) is not a holiday, please fix

**Comment by brunobord** [comment]
Hello.

Thank you for raising this issue. Do you have more information on this matter? As far as I can tell, hen HK calendar was added to the library, there was a shift (otherwise, I don't think it would have been added).
Maybe we could handle and fix this issue if we had an _"official source of truth"_ to make sure that this calendar is correct for every year supported?

Many thanks.

**Comment by greatfung** [comment]
Hi Bruno,

I can only get you the latest cap 149 which was last updated on 2012.
https://www.elegislation.gov.hk/hk/cap149!en-zh-Hant-HK.pdf

I thought the shift had been brought up for public discussion several years
ago, but it has not been brought up to legislative council yet, thus not
yet approved.
So for 2020, you can see from the official holiday list
https://www.gov.hk/en/about/abouthk/holiday/2020.htm, Dec 28, 2020 is not a
holiday.

In short, Sunday shift YES, Saturday shift NO
Hope this helps.

Regards,
kimfung

**Comment by brunobord** [comment]
It looks like the problem is deeper that I thought.
The most interesting page of the official document, as I can see is the one declaring the "General Holidays". Reproducing its English content here:

> (a) every Sunday;
> (b) the first day of January (or if that day is a Sunday, then the following day);
> (c) Lunar New Year’s Day (or if that day is a Sunday, then the fourth day of Lunar New Year); (Amended 23 of 2011 s. 3)
> (d) the second day of Lunar New Year (or if that day is a Sunday, then the fourth day of Lunar New Year); (Amended 23 of 2011 s. 3)
> (e) the third day of Lunar New Year (or if that day is a Sunday, then the fourth day of Lunar New Year); (Amended 23 of 2011 s. 3)
> (f) Ching Ming Festival (or if that day is a Sunday, then the following day);
> (g) Good Friday;
> (h) the day following Good Friday;
> (i) Easter Monday;
> (j) Labour Day, being the first day of May (or if that day is a Sunday, then the following day);
> (k) the Birthday of the Buddha, being the eighth day of the fourth lunar month (or if that day is a Sunday, then the following day);
> (l) Tuen Ng Festival (or if that day is a Sunday, then the following day);
> (m) Hong Kong Special Administrative Region Establishment Day, being the first day of July (or if that day is a Sunday, then the following day);
> (n) National Day, being the first day of October (or if that day is a Sunday, then the following day);
> (o) the day following the Chinese Mid-Autumn Festival (or if that day is a Sunday, then the second day following that Festival) or such other day as the Chief Executive in Council may, by order in the Gazette, appoint in place of that day; (Amended 23 of 2011 s. 3)
> (p) Chung Yeung Festival (or if that day is a Sunday, then the following day) or such other day as the Chief Executive in Council may, by order in the Gazette, appoint in place of that day;
> (q) Christmas Day (or if that day is a Sunday, then the second weekday after Christmas Day);
> (r) the first weekday after Christmas Day.

----

Here is a **big** question: I see that holidays are only "Sundays". Does that mean that every Saturday has to be counted as a working day?

**Comment by greatfung** [comment]
Yes Bruno, Saturdays are working day in HK, however for banks, financial
institutions, large companies or government, most of the departments are
carrying 5-day workweek, only limited service will be provided, that's why
you have an impression that Saturdays are holiday.

**Comment by greatfung** [comment]
But if this is a calendar for financial use, you may treat Saturdays as
holiday. However, any public holiday fall into Saturday will not be
shifted.

**Comment by brunobord** [comment]
So maybe the solution would be to have a "normal" calendar with Saturdays as working holidays, and a "Bank" calendar including saturdays as non-working days. And both would shift a holiday if it falls on a Sunday.

Sounds like a plan. It'll take some time, but I'm glad we would have a more solid HK calendar.
Stay tuned!

**Comment by brunobord** [comment]
Release is on its way, stay tuned.

**Comment by brunobord** [comment]
Version 8.3.0 was released and pushed on PyPI. Happy upgrading!

* https://github.com/peopledoc/workalendar/releases/tag/8.3.0
* https://pypi.org/project/workalendar/
